lib/document.js

```diff
@@ -809,6 +809,12 @@ Document.prototype._onSaved = function(result, docToSave, saveAll, savedModel, r
   var self = this;
 
   if (result.first_error != null) {
+    var duplicatePrimaryKeyRegex = new RegExp('^' + new Errors.DuplicatePrimaryKey().name);
```

Comment:
Can you move this in errors.js? And add a `Error.create` method that takes a string and return the appropriate error type.
That would let us keep all the logic related to errors in the same places and would make adding new error type easier.
Thanks!